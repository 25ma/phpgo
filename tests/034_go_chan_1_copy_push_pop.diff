007+ pushed true
008- pushed true