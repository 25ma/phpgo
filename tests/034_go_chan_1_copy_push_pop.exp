SUB-TC: #1
push
pushed abc
pop
popped:string(3) "abc"
pushed 100
popped:int(100)
pushed true
popped:bool(true)
pushed false
popped:bool(false)