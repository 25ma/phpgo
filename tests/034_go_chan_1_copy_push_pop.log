
---- EXPECTED OUTPUT
SUB-TC: #1
push
pushed abc
pop
popped:string(3) "abc"
pushed 100
popped:int(100)
pushed true
popped:bool(true)
pushed false
popped:bool(false)
---- ACTUAL OUTPUT
SUB-TC: #1
push
pushed abc
pop
popped:string(3) "abc"
pushed 100
pushed true
popped:int(100)
popped:bool(true)
pushed false
popped:bool(false)
---- FAILED
