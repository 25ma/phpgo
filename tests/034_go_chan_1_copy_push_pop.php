<?php
use \go\Chan;
use \go\Scheduler;

function subtc($seq){
    echo "SUB-TC: #$seq\n";
}

subtc(1);
$ch = new Chan(["capacity"=>1, "copy"=>true]);

go(function($ch){
	echo "push\n";
	$v = "abc";
	$ch->push($v);
	echo "pushed $v\n";
	$ch->push(100);
	echo "pushed 100\n";
	$ch->push(true);
	echo "pushed true\n";
	$ch->push(false);
	echo "pushed false\n";
},$ch);

go(function($ch){
	echo "pop\n";
	$v = $ch->pop();
	echo "popped:";
	var_dump($v);
	$v = $ch->pop();
	echo "popped:";
	var_dump($v);
	$v = $ch->pop();
	echo "popped:";
	var_dump($v);
	$v = $ch->pop();
	echo "popped:";
	var_dump($v);
},$ch);

Scheduler::RunJoinAll();

?>
