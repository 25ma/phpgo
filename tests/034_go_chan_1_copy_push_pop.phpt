--TEST--
Go Chan(1,copy) push pop

--FILE--
<?php
use \go\Chan;
use \go\Scheduler;

function subtc($seq){
    echo "SUB-TC: #$seq\n";
}

subtc(1);
$ch = new Chan(["capacity"=>1, "copy"=>true]);

go(function($ch){
	$v = "abc";
	
	echo "go1: try push $v\n";
	$ch->push($v);
	echo "go1: pushed $v\n";
	
	echo "go1: try push 100\n";
	$ch->push(100);
	echo "go1: pushed 100\n";
	
	echo "go1: try push true\n";
	$ch->push(true);
	echo "go1: pushed true\n";
	
	echo "go1: try push false\n";
	$ch->push(false);
	echo "go1: pushed false\n";
},$ch);

go(function($ch){
	echo "go2: try pop\n";
	$v = $ch->pop();
	echo "go2: popped:";	var_dump($v);
	
	echo "go2: try pop\n";
	$v = $ch->pop();
	echo "go2: popped:";	var_dump($v);
	
	echo "go2: try pop\n";
	$v = $ch->pop();
	echo "go2: popped:";	var_dump($v);
	
	echo "go2: try pop\n";
	$v = $ch->pop();
	echo "go2: popped:";	var_dump($v);
},$ch);

Scheduler::join();

?>
--EXPECT--
SUB-TC: #1
go1: try push abc
go1: pushed abc
go1: try push 100
go2: try pop
go2: popped:string(3) "abc"
go2: try pop
go1: pushed 100
go1: try push true
go1: pushed true
go1: try push false
go2: popped:int(100)
go2: try pop
go2: popped:bool(true)
go2: try pop
go1: pushed false
go2: popped:bool(false)
