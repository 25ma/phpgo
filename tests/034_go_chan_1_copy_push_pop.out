SUB-TC: #1
push
pushed abc
pop
popped:string(3) "abc"
pushed 100
pushed true
popped:int(100)
popped:bool(true)
pushed false
popped:bool(false)